### ISSUE TITLE
let/const scope is calculated incorrectly

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#917]
The following valid code:
```js
/* global a: false */
if (a) {
	let b = 2;
} else {
	let b = 3;
}
```
triggers the `b is already defined` error in the `no-redeclare` rule.

**Comment by nzakas** [comment]
Working through some issues with escope that will address this in #1642

**Comment by nzakas** [comment]
Working on this.
